beater/handlers.go

```diff
@@ -342,11 +350,10 @@ func sendStatus(w http.ResponseWriter, r *http.Request, code int, err error) {
 	}
 
 	logger, ok := r.Context().Value(reqLoggerContextKey).(*logp.Logger)
-	if ok {
-		logger.Errorw("error handling request", "error", err.Error())
-	} else {
-		logp.NewLogger("request").Errorf("error handling request: %v", err)
+	if !ok {
+		logger = logp.NewLogger("request")
 	}
+	logger.Errorw("error handling request", "error", err.Error())
```

Comment:
Shouldn't this be inside the if-clause?

Reply:
This if clause only checks if there is a valid logger given in the context, otherwise a new logger is created. In both cases the error that occured should be logged. 
This code is only reached if there was an error, otherwise it returns here https://github.com/elastic/apm-server/pull/811/files/3db639d45486fc0d2700741512af430c312e81b7#diff-0ee6b4dd463eb95280cc3653bd4701b6R348.

Reply:
ah right. I'll retract my comment then!